### PR TITLE
DOC Link `magicgui` type annotation to providers/processors

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues

See https://github.com/napari/napari/pull/4543 for Talley original PR and his notes on this.

# Description

Links `magicgui` type annotation and providers/processor behaviour.

### CODE REVIEW

**Comment by lucyleeow** [comment]
cc @DragaDoncila

File: docs/developers/architecture/app_model.md [hunk 1]
```diff
@@ -477,6 +477,16 @@ They would be registered in the `app` `Store` during
 initialization of the napari `app`, in `NapariApplication`'s
 {meth}`~napari._app_model._app.NapariApplication.__init__`.
 
+Providers and processors allow us to mirror behaviour enabled by
+[`magicgui` type registration](magicgui_type_registration) for widgets. Providers
+allow generalization of `magicgui` parameter annotation behaviour to all
+actions, while processors does this for return annotation behaviour.
```
Comment (DragaDoncila) [inline docs/developers/architecture/app_model.md]:
```suggestion
actions, while processors do this for return annotation behaviour.
```

File: docs/developers/architecture/app_model.md [hunk 2]
```diff
@@ -477,6 +477,16 @@ They would be registered in the `app` `Store` during
...
+
+The intended users of providers/processors are end-users, plugins as well as `napari`
+internally.
+For a detailed discussion on this framework see
+[issue 4532](https://github.com/napari/napari/issues/4532).
```
Comment (DragaDoncila) [inline docs/developers/architecture/app_model.md:488]: Maybe a more explicit note that currently it's only really used internally?

**Review by DragaDoncila — APPROVED** [review]
Approved with two small comments, thanks Lucy!

File: docs/developers/architecture/app_model.md (hunk 2 above)
Comment (lucyleeow) [reply docs/developers/architecture/app_model.md:488]: Amended, let me know if its okay!

Comment (DragaDoncila) [reply docs/developers/architecture/app_model.md:488]: yep, added ready to merge!

**Comment by melissawm** [comment]
Thanks @lucyleeow and @DragaDoncila !

**Pull request merged**